articles/sorting-algorithms-in-js/index.md

```diff
@@ -0,0 +1,508 @@
+            i++;
+        }else {
+            res.push(arr2[j]);
+            j++;
+        }
+    };
+
+    //pushing the rest of arr1.
+    while(i < arr1.length){
+        res.push(arr1[i]);
+        i++;
+    };
+
+    //pushing the rest of arr2.
+    while(j < arr2.length){
+        res.push(arr2[j]);
+        j++;
+    }
+
+    return res;
+};
+
+//merge sort
+function mergeSort(arr){
+
+    //Best case 
+    if(arr.length <= 1) return arr;
+
+    //splitting into halves
+    let mid = Math.ceil(arr.length / 2);
+
+    let arr1 = arr.slice(0,mid);
+
+    let arr2 = arr.slice(mid);
+
+    let arr1_sublists = [],sorted_arr1_sublists = [];
+
+    let arr2_sublists = [],sorted_arr2_sublists = [];
+
+    //loop through array 1 making sublists of two elements
+    for(let i = 0; i < arr1.length; i+=2){
+        arr1_sublists.push(arr1.slice(i,i + 2));
+    };
+
+    //loop through array 2 making sublists of two elements.
+    for(let i = 0; i < arr2.length; i+=2){
+        arr2_sublists.push(arr2.slice(i,i + 2));
+    };
+
+    // sorting each sublist of arr1.
+    for(let i = 0; i < arr1_sublists.length; i+=2){
+        let result = merge(arr1_sublists[i],arr1_sublists[i + 1]);
+        result.forEach((value) => sorted_arr1_sublists.push(value));
+    };
+
+    // sorting each sublist of arr2.
+    for(let i = 0; i < arr2_sublists.length; i+=2){
+        let result = merge(arr2_sublists[i],arr2_sublists[i + 1]);
+        result.forEach((value) => sorted_arr2_sublists.push(value));
+    };
+
+
+    let result = merge(sorted_arr1_sublists,sorted_arr2_sublists);
+
+    return result;
+};
+
+console.log(mergeSort([70,50,30,10,20,40,60]));
+```
+
+Output
+
+```bash
+[
+  10, 20, 30, 40,
+  50, 60, 70
+]
+```
+
+Merge sort has the following performance cases:
+
+- Worst-case time complexity: [`Big O`](https://www.geeksforgeeks.org/difference-between-big-oh-big-omega-and-big-theta/) n * log n. It divides a list into two halves and takes linear time to merge the two halves.
+
+- Average-case time complexity: [`Big theta`](https://www.geeksforgeeks.org/difference-between-big-oh-big-omega-and-big-theta/) n * log n. It divides a list into two halves and takes linear time to merge the two halves.
+
+- Best-case time complexity: [`Big omega`](https://www.geeksforgeeks.org/difference-between-big-oh-big-omega-and-big-theta/) n * log n. It divides a list into two halves and takes linear time to merge the two halves. 
+
+- Space complexity: [`Big-O`](https://www.geeksforgeeks.org/difference-between-big-oh-big-omega-and-big-theta/) n. It requires additional space based on the elements of the list. It is not preferred for large lists.
+
+### Quick sort
+
+Quicksort applies the divide and conquer technique. It works by having a pivot element such that the elements to the left of it are less than it and those to the right are greater than it. The pivot element can be any element in the list.
+
+**Step-by-step guide**:
+
+- Identify the pivot element.
```

Comment:
`Identify` is misleading, you can say select a pivot element.

Reply:
fixed.